gcloud/bigquery/job.py

```diff
@@ -23,6 +23,121 @@
 from gcloud.bigquery.table import _parse_schema_resource
 
 
+class _ConfigurationProperty(object):
+    """Base property implementation.
+
+    Values will be stored on a `_configuration` helper attribute of the
+    property's job instance.
+
+    :type name: string
+    :param name:  name of the property
+    """
+
+    def __init__(self, name):
+        self.name = name
+        self._backing_name = '_%s' % (self.name,)
+
+    def __get__(self, instance, owner):
+        """Descriptor protocal:  accesstor"""
+        if instance is None:
+            return self
+        return getattr(instance._configuration, self._backing_name)
+
+    def _validate(self, value):
```

Comment:
If we're going to do this, we should just dive in head first and use enums (not for the descriptor behavior, just the validation). As I read on, I may have more to say / erase this.